### PR TITLE
Add missing NoOp case to ToTable

### LINKED ISSUE #519: Shuffle(Table) doesn't work for Lazy Table
Given a table of a custom lazy type:

```
class LazyTableType : TableType
{
  //...
}
```

Calling `Filter(Table, ...)` works fine but `Shuffle(Table)` complains about argument types. It's possible that other table functions also have this issue, I haven't done an exhaustive search.

Possibly this has to do with [`ShuffleFunction.SupportsParamCoercion` being false](https://github.com/microsoft/Power-Fx/blob/916b15d6a4e2c7915004acf3ef31ec1e58e46c98/src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Shuffle.cs#L22)?

### PR DESCRIPTION
**Pull request by lesaltzm**
fixes #519

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — DISMISSED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs [hunk 1]
```diff
@@ -1087,6 +1087,7 @@ public DType ToTable(ref bool fError)
 
             switch (Kind)
             {
+                case DKind.LazyTable:
                 case DKind.Table:
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:1091]: >  case DKind.Table: [](http://example.com/codeflow?start=14&length=19)

This looks good - can we check other cases in this file (or does PAClient have other cases too?)

**Comment by MikeStall** [comment]
>                 case DKind.Table:

here?

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:1247 in 2d126fb. [](commit_id = 2d126fb0e0f1460b98d103c316d4023c208eb61a, deletion_comment = False)

**Comment by MikeStall** [comment]
>                 case DKind.Table:

here?

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:1679 in 2d126fb. [](commit_id = 2d126fb0e0f1460b98d103c316d4023c208eb61a, deletion_comment = False)

**Comment by MikeStall** [comment]
>                 case DKind.Table:

here?

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:1796 in 2d126fb. [](commit_id = 2d126fb0e0f1460b98d103c316d4023c208eb61a, deletion_comment = False)

**Comment by MikeStall** [comment]
>                 case DKind.Table:                    

here?

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:1913 in 2d126fb. [](commit_id = 2d126fb0e0f1460b98d103c316d4023c208eb61a, deletion_comment = False)

**Comment by MikeStall** [comment]
>                 case DKind.Table:

here?

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:2943 in 2d126fb. [](commit_id = 2d126fb0e0f1460b98d103c316d4023c208eb61a, deletion_comment = False)

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**